src/test/ensemble_metadata_test.cpp

```diff
@@ -347,3 +348,315 @@ TEST(EnsembleMetadata, OneUnavailableNode) {
+        (*info)[1].dims[1] = 400;
+        return 0;
+    }
+    static int getOutputsInfo(struct CustomNodeTensorInfo** info, int* infoLength, const struct CustomNodeParam*, int) {
+        *infoLength = 2;
+        *info = (struct CustomNodeTensorInfo*)malloc(*infoLength * sizeof(struct CustomNodeTensorInfo));
+
+        (*info)[0].name = "output_A_1";
+        (*info)[0].precision = FP32;
+        (*info)[0].dimsLength = 3;
+        (*info)[0].dims = (uint64_t*)malloc((*info)[0].dimsLength * sizeof(uint64_t));
+        (*info)[0].dims[0] = 1;
+        (*info)[0].dims[1] = 3;
+        (*info)[0].dims[2] = 10;
+
+        (*info)[1].name = "output_B_1";
+        (*info)[1].precision = FP32;
+        (*info)[1].dimsLength = 3;
+        (*info)[1].dims = (uint64_t*)malloc((*info)[0].dimsLength * sizeof(uint64_t));
+        (*info)[1].dims[0] = 1;
+        (*info)[1].dims[1] = 3;
+        (*info)[1].dims[2] = 4;
+        return 0;
+    }
+    static int release(void* ptr) {
+        free(ptr);
+        return 0;
+    }
+};
+
+struct LibraryDemultiplexer_2 {
+    static int execute(const struct CustomNodeTensor*, int, struct CustomNodeTensor**, int*, const struct CustomNodeParam*, int) {
+        return 1;
+    }
+    static int getInputsInfo(struct CustomNodeTensorInfo** info, int* infoLength, const struct CustomNodeParam*, int) {
+        *infoLength = 2;
+        *info = (struct CustomNodeTensorInfo*)malloc(*infoLength * sizeof(struct CustomNodeTensorInfo));
+
+        (*info)[0].name = "input_A_2";
+        (*info)[0].precision = FP32;
+        (*info)[0].dimsLength = 2;
+        (*info)[0].dims = (uint64_t*)malloc((*info)[0].dimsLength * sizeof(uint64_t));
+        (*info)[0].dims[0] = 1;
+        (*info)[0].dims[1] = 10;
+
+        (*info)[1].name = "input_B_2";
+        (*info)[1].precision = FP32;
+        (*info)[1].dimsLength = 2;
+        (*info)[1].dims = (uint64_t*)malloc((*info)[0].dimsLength * sizeof(uint64_t));
+        (*info)[1].dims[0] = 1;
+        (*info)[1].dims[1] = 4;
+        return 0;
+    }
+    static int getOutputsInfo(struct CustomNodeTensorInfo** info, int* infoLength, const struct CustomNodeParam*, int) {
+        *infoLength = 1;
+        *info = (struct CustomNodeTensorInfo*)malloc(*infoLength * sizeof(struct CustomNodeTensorInfo));
+
+        (*info)[0].name = "output_A_2";
+        (*info)[0].precision = FP32;
+        (*info)[0].dimsLength = 3;
+        (*info)[0].dims = (uint64_t*)malloc((*info)[0].dimsLength * sizeof(uint64_t));
+        (*info)[0].dims[0] = 1;
+        (*info)[0].dims[1] = 4;
+        (*info)[0].dims[2] = 10;
+        return 0;
+    }
+    static int release(void* ptr) {
+        free(ptr);
+        return 0;
+    }
+};
+
+TEST(EnsembleMetadata, CustomNodeMultipleDemultiplexers) {
+    ConstructorEnabledModelManager manager;
+    ModelConfig config = DUMMY_MODEL_CONFIG;
+    ASSERT_EQ(manager.reloadModelWithVersions(config), StatusCode::OK);
+
+    NodeLibrary library_1{
```

Comment:
Same with library names here -> point which one has how much inputs/outputs.

Reply:
I don't think we really care about numbers of inputs/outputs more than actual metadata.